dev/ha/config.go

```diff
@@ -0,0 +1,632 @@
+				ans.ElectionTimersMode = ElectionTimersModeAdvanced
+				x := o.ElectionOption.Timers.Advanced
+				ans.ElectionTimersAdvPromotionHoldTime = x.PromotionHoldTime
+				ans.ElectionTimersAdvHelloInterval = x.HelloInterval
+				ans.ElectionTimersAdvHeartBeatInterval = x.HeartBeatInterval
+				ans.ElectionTimersAdvFlapMax = x.FlapMax
+				ans.ElectionTimersAdvPreemptionHoldTime = x.PreemptionHoldTime
+				ans.ElectionTimersAdvMonitorFailHoldUpTime = x.MonitorFailHoldUpTime
+				ans.ElectionTimersAdvAdditionalMasterHoldUpTime = x.AdditionalMasterHoldUpTime
+			}
+		}
+	}
+
+	if o.StateSync != nil {
+		ans.Ha2StateSyncEnable = util.AsBool(o.StateSync.Enable)
+		ans.Ha2StateSyncTransport = o.StateSync.Transport
+		ans.Ha2StateSyncKeepAliveEnable = util.AsBool(o.StateSync.Ha2KeepAlive.Enable)
+		ans.Ha2StateSyncKeepAliveAction = o.StateSync.Ha2KeepAlive.Action
+		ans.Ha2StateSyncKeepAliveThreshold = o.StateSync.Ha2KeepAlive.Threshold
+	}
+
+	if o.LinkMonitor != nil {
+		ans.LinkMonitorEnable = util.AsBool(o.LinkMonitor.Enable)
+		ans.LinkMonitorFailureCondition = o.LinkMonitor.FailureCondition
+		if o.LinkMonitor.LinkGroup != nil {
+			raw["linkgroup"] = util.CleanRawXml(o.LinkMonitor.LinkGroup.Text)
+		}
+	}
+
+	if o.Cluster != nil {
+		raw["cluster"] = util.CleanRawXml(o.Cluster.Text)
+	}
+	if o.PathMonitor != nil {
+		raw["pathmonitor"] = util.CleanRawXml(o.PathMonitor.Text)
+	}
+
+	if len(raw) != 0 {
+		ans.raw = raw
+	}
+
+	return ans
+}
+
+type entry_v1 struct {
+	XMLName                xml.Name        `xml:"high-availability"`
+	Enable                 string          `xml:"enabled"`
+	Interfaces             *interfaces     `xml:"interface"`
+	GroupId                int             `xml:"group>group-id,omitempty"`
+	Description            string          `xml:"group>description,omitempty"`
+	Mode                   *mode           `xml:"group>mode"`
+	PeerHa1IpAddress       string          `xml:"group>peer-ip,omitempty"`
+	BackupPeerHa1IpAddress string          `xml:"group>peer-ip-backup,omitempty"`
+	ConfigSyncEnable       string          `xml:"group>configuration-synchronization>enabled"`
+	ElectionOption         *electionOption `xml:"group>election-option"`
+	StateSync              *stateSync      `xml:"group>state-synchronization"`
+	LinkMonitor            *linkMonitor    `xml:"group>monitoring>link-monitoring"`
+
+	PathMonitor *util.RawXml `xml:"group>monitoring>path-monitoring"`
+	Cluster     *util.RawXml `xml:"cluster"`
+}
+
+type interfaces struct {
+	Ha1       *ha1Interface       `xml:"ha1"`
+	Ha1Backup *ha1BackupInterface `xml:"ha1-backup"`
+	Ha2       *ha2Interface       `xml:"ha2"`
+	Ha2Backup *ha2Interface       `xml:"ha2-backup"`
+	Ha3       *ha3Interface       `xml:"ha3"`
+	Ha4       *ha4Interface       `xml:"ha4"`
+	Ha4Backup *ha4Interface       `xml:"ha4-backup"`
+}
+
+type common struct {
+	Port      string `xml:"port,omitempty"`
+	IpAddress string `xml:"ip-address,omitempty"`
+	Netmask   string `xml:"netmask,omitempty"`
+}
```

Comment:
Please ditch this anonymous struct; just put the params in the structs that need them.